docs/conf.py

```diff
@@ -45,6 +49,26 @@
     "sphinx.ext.inheritance_diagram",
 ]
 
+# Modules needed to import django files
+sys.modules["psycopg.types"] = Mock()
+# These modules run code on import which are dependent on other packages
+sys.modules["django.db.backends.postgresql.psycopg_any"] = Mock()
+sys.modules["django.contrib.gis.geos"] = Mock()
+sys.modules["django.contrib.gis.gdal.libgdal"] = Mock()
+sys.modules["django.contrib.gis.gdal.prototypes"] = Mock()
+sys.modules["django.contrib.gis.db.models.functions"] = Mock()
+# Load django apps for generating their source by 'sphinx.ext.viewcode'
```

Comment:
It's not clear to me from the comment why some apps are listed and not others.

Reply:
I updated the comments :+1: let me know if I should expand them further

Reply:
I'm not seeing the comment changes. My question is, for example, why is "django.contrib.admin" listed but not contrib.sessions? Also, why do some dependencies like psycopg need to be mocked but not others like the drivers for other databases?

Reply:
My process for deciding what imports to mock and what apps to include is as following:
- Update the `html` command to be run with [increased verbosity](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-v) (`-v`)
- Run `make clean html` 
- Check the logs

The logs are very busy and I was mostly looking for:
```
viewcode can't import [some Django file], failed with error "No module named '[some missing dependency or app]'"
```
There are also errors like: `Didn't find [something] in [some Django file]` and so I added a few more apps and these decreased.

With each of the mocked dependencies or added apps, you should find that if you remove it, the logs increase (and possibly there are less linked modules).

 ----

There are many things that can still be fixed in the logs and there are multiple reasons being why Sphinx is unable to find the reference, but I'm hoping the PR will fix the external dependency issues and needed Django apps.

I'm struggling to formulate this into a useful explanatory comment but hopefully this is clearer on what I am doing.

Reply:
Tried to update the comment to be clearer :+1: